client/dashboard/app/queries/domain-nameservers.ts

```diff
@@ -0,0 +1,44 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { fetchDomainNameservers, updateDomainNameservers } from '../../data/domain-nameservers';
+
+const useDomainNameserversQuery = ( domainName: string ) =>
```

Comment:
In the new dashboard code, we only specify query options in the `app/queries/` layer. Then, we call `useQuery()` in the component instead. Could we follow this pattern for consistency?

Similarly for mutations.